bundles/org.openhab.voice.googlestt/src/main/java/org/openhab/voice/googlestt/internal/GoogleSTTService.java

```diff
@@ -0,0 +1,381 @@
+        private float confidenceSum = 0;
+        private int responseCount = 0;
+        private long lastInputTime = 0;
+
+        public TranscriptionListener(STTListener sttListener, GoogleSTTConfiguration config,
+                Consumer<@Nullable Throwable> completeListener) {
+            this.sttListener = sttListener;
+            this.config = config;
+            this.completeListener = completeListener;
+        }
+
+        public void onStart(@Nullable StreamController controller) {
+            sttListener.sttEventReceived(new SpeechStartEvent());
+            lastInputTime = System.currentTimeMillis();
+        }
+
+        public void onResponse(StreamingRecognizeResponse response) {
+            lastInputTime = System.currentTimeMillis();
+            List<StreamingRecognitionResult> results = response.getResultsList();
+            logger.debug("Got {} results", response.getResultsList().size());
+            if (results.isEmpty()) {
+                logger.debug("No results");
+                return;
+            }
+            results.forEach(result -> {
+                List<SpeechRecognitionAlternative> alternatives = result.getAlternativesList();
+                logger.debug("Got {} alternatives", alternatives.size());
+                SpeechRecognitionAlternative alternative = alternatives.stream()
+                        .min(Comparator.comparing(SpeechRecognitionAlternative::getConfidence)).orElse(null);
+                if (alternative == null) {
+                    return;
+                }
+                String transcript = alternative.getTranscript();
+                logger.debug("Alternative transcript: {}", transcript);
+                logger.debug("Alternative confidence: {}", alternative.getConfidence());
+                if (result.getIsFinal()) {
+                    transcriptBuilder.append(transcript);
+                    confidenceSum += alternative.getConfidence();
+                    responseCount++;
+                    // when in single utterance mode we can just get one final result so complete
+                    if (config.singleUtteranceMode) {
+                        completeListener.accept(null);
+                    }
+                }
+            });
+        }
+
+        public void onComplete() {
+            sttListener.sttEventReceived(new SpeechStopEvent());
+            float averageConfidence = confidenceSum / (float) responseCount;
+            String transcript = transcriptBuilder.toString();
+            if (!transcript.isBlank()) {
+                sttListener.sttEventReceived(new SpeechRecognitionEvent(transcript, averageConfidence));
+            } else {
+                if (!config.noResultsMessage.isBlank()) {
+                    sttListener.sttEventReceived(new SpeechRecognitionErrorEvent(config.noResultsMessage));
+                } else {
+                    sttListener.sttEventReceived(new SpeechRecognitionErrorEvent("No results"));
```

Comment:
You have to put this text in properties to make it translatable in other languages.
If I am not wrong, this message will be said with the audio sink.

Reply:
The user can overwrite this message. Is it not enough?
If not, can you point me to an example? I have no idea how to do that.

Reply:
You're right. 
Keep it like that.